### PR TITLE
doc: mention exception for JoinGuard.detach() behavior

### PR DESCRIPTION
**Pull request by tshepang**

### CODE REVIEW

File: src/libstd/thread.rs [hunk 1]
```diff
@@ -416,6 +416,8 @@ impl<T: Send> JoinGuard<T> {
     }
 
     /// Detaches the child thread, allowing it to outlive its parent.
+    /// Note, however, that the main thread is special in that if it quits,
+    /// all other threads are also going to quit.
```
Comment (cmr) [inline src/libstd/thread.rs]: I think it's misleading to say that the other threads quit, but rather just that the program terminates.

**Comment by steveklabnik** [comment]
Can you squash please? r=me after

**Comment by tshepang** [comment]
Closing in favor of https://github.com/rust-lang/rust/pull/20635.
